app/controllers/declaration/AdditionalDeclarationTypeController.scala

```diff
@@ -48,15 +51,21 @@ class AdditionalDeclarationTypeController @Inject()(
   }
 
   def submitForm(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val decType = extractFormType(request)
+    val action = FormAction.bindFromRequest
+    val decType = extractFormType(request).form().bindFromRequest()
+
     decType
-      .form()
-      .bindFromRequest()
       .fold(
         formWithErrors => Future.successful(BadRequest(declarationTypePage(formWithErrors))),
         validAdditionalDeclarationType =>
-          updateCache(validAdditionalDeclarationType)
-            .map(_ => Redirect(controllers.declaration.routes.ConsignmentReferencesController.displayPage()))
+          updateCache(validAdditionalDeclarationType).map { _ =>
+            action match {
```

Comment:
What if action is None ?

What do you think about link navigator and action together ?

Reply:
Action = None means you've hacked the UI, i'm not sure we are too bothered :)

Reply:
So maybe we should move away from `Option` here - if it is meaningless why we should bother about it.

Reply:
Good question, one for the tech debt board